src/main/java/com/google/devtools/build/lib/remote/options/RemoteOptions.java

```diff
@@ -127,6 +127,39 @@ concurrent requests (controlled by --remote_max_concurrency_per_connection), so
               + " set.")
   public boolean remoteCacheAsync;
 
+  /** Determines the mode for uploading outputs to the remote cache. */
+  public enum RemoteUploadMode {
+    /** Block at the end of the build waiting for all uploads to complete (current behavior). */
+    WAIT_FOR_UPLOAD_COMPLETE,
+    /** Block at the beginning of the next build waiting for upload completion. */
+    NOWAIT_FOR_UPLOAD_COMPLETE,
+  }
+
+  /** Converter for {@link RemoteUploadMode}. */
+  public static class RemoteUploadModeConverter extends EnumConverter<RemoteUploadMode> {
+    public RemoteUploadModeConverter() {
+      super(RemoteUploadMode.class, "remote upload mode");
+    }
+  }
+
+  @Option(
+      name = "experimental_remote_upload_mode",
```

Comment:
Instead of adding a new flag, how about turning `--remote_cache_async` into an enum-valued flag with three states? A `BoolOrEnumConverter` would preserve backwards compatibility.